62.CAD/main.cpp

```diff
@@ -478,14 +1464,62 @@ struct DrawBuffersFiller
 		return intendedNextSubmit;
 	}
 
-	// TODO[Lucas]: drawHatch function with similar signature to drawPolyline
 	// If we had infinite mem, we would first upload all curves into geometry buffer then upload the "CurveBoxes" with correct gpu addresses to those
 	// But we don't have that so we have to follow a similar auto submission as the "drawPolyline" function with some mutations:
 	// We have to find the MAX number of "CurveBoxes" we could draw, and since both the "Curves" and "CurveBoxes" reside in geometry buffer,
 	// it has to be taken into account when calculating "how many curve boxes we could draw and when we need to submit/clear"
 	// So same as drawPolylines, we would first try to fill the geometry buffer and index buffer that corresponds to "backfaces or even provoking vertices"
 	// then change index buffer to draw front faces of the curveBoxes that already reside in geometry buffer memory
 	// then if anything was left (the ones that weren't in memory for front face of the curveBoxes) we copy their geom to mem again and use frontface/oddProvoking vertex
+	video::IGPUQueue::SSubmitInfo drawHatch(
+		const Hatch& hatch,
+		const CPULineStyle& lineStyle,
```

Comment:
you use it only for the color right? change this "input" to a float32_t4 color, if you use it for more than a color then make a new HatchStyle or something and take that as input.
Then convert to a line style internally. cause we may store it in a different place in the gpu later on without needing to touch the api and usage of this function everywhere else